### PR TITLE
stroke width calculation

### PR DESCRIPTION
**Pull request by kaalleen**
If we convert lines to satin columns we possibly see a huge difference between the visual width in the Inkscape canvas compared to the width of the final satin column. This happens because we do not calculate any transforms on top of the stroke-width setting other than doc size and viewbox. We can not be fully accurate if the scaling in x and y directions are not equal.

### CODE REVIEW

**Comment by lexelby** [comment]
THANK YOU for working on this.  This bug has bothered me and lots of users, especially in the French facebook group.

File: lib/elements/element.py [hunk 1]
```diff
@@ -186,15 +189,18 @@ def has_style(self, style_name):
...
+        if node_transform is not None:
+            if node_transform.startswith('matrix'):
+                decomposed_transform = findall(r"[-+]?\d*\.\d+|\d+", node_transform)
+                sx = sqrt(float(decomposed_transform[0])**2 + float(decomposed_transform[2])**2)
+                sy = sqrt(float(decomposed_transform[1])**2 + float(decomposed_transform[3])**2)
+                node_scale = (sx + sy) / 2
+            elif node_transform.startswith('scale'):
+                decomposed_transform = findall(r"[-+]?\d*\.\d+|\d+", node_transform)
+                node_scale = (float(decomposed_transform[0]) + float(decomposed_transform[1])) / 2
+        return node_scale
```
Comment (lexelby) [inline lib/elements/element.py:203]: Hmmmmm.  I wonder if we can get creative here.  What if we do `node_transform.apply_to_point((1, 0))`, then take the magnitude of the resulting vector to get `sx`?  Then repeat for `(0, 1)` to get `sy`.  I think we may be able to get inkex to do all the work for us.  What do you think?

Comment (lexelby) [reply lib/elements/element.py:203]: Oh, my code only works for inkex 1.0+, but there's a similar function to apply a transform to a point in the old inkex

**Comment by kaalleen** [comment]
> THANK YOU for working on this. This bug has bothered me and lots of users, especially in the French facebook group.

Oh, I had no idea. @LaureFR38 pointed me to it. So the credit is hers ;)
I just joined the facebook groups recently, two or three days ago.

File: lib/elements/element.py (hunk 1 above)
Comment (kaalleen) [reply lib/elements/element.py:203]: I don't know. Maybe this goes a little bit above my head. ... could you take over?
My code here is working, but it's not really very beautiful. I do admit that :D

Comment (lexelby) [reply lib/elements/element.py:203]: Sure thing. :)  This is something that will probably make more sense in code than English.

**Comment by lexelby** [comment]
Coded on my phone and not tested yet -- hopefully I got it right. :)

I noticed that you used `str()` on a transform which I think only works in inkex 1.0+?  But it looks like this branch is based on `main`, not your Python 3.0 branch?

**Comment by kaalleen** [comment]
Nice, I see where it is going. Without testing it I would think we need to import Point and change the second sx to be sy?

**Comment by kaalleen** [comment]
> I noticed that you used str() on a transform which I think only works in inkex 1.0+?

Right. I am all along working with Inkscape 1.0+ now. So I am getting confused about old and new stuff.

**Comment by lexelby** [comment]
Correct, I had a few bugs. :) The most important of which was that I needed to also cancel out the translation component of the transform.  This version seems to do the job nicely.

**Comment by kaalleen** [comment]
Interesting approach. I would never have come up with something like that. I am also not sure if I understand every detail of what it's doing. But it gets the job done 👍

**Review by lexelby — APPROVED** [review]

**Pull request merged**